client/my-sites/plan-features/index.jsx

```diff
@@ -585,6 +594,13 @@ export class PlanFeatures extends Component {
 				'has-border-bottom',
 				'is-bottom-buttons'
 			);
+
+			if ( disableBloggerPlanWithNonBlogDomain || this.props.nonDotBlogDomains.length > 0 ) {
+				if ( planMatches( planName, { type: TYPE_BLOGGER } ) ) {
+					availableForPurchase = false;
+				}
```

Comment:
Do you think it would make sense to do this in `mapStateToProps` by setting `availableForPurchase` through a function rather than a ternary?

Reply:
Seems like it would be the same to me, but happy to change it if you feel strongly :+1: